bika/lims/browser/js/coffee/bika.lims.site.coffee

```diff
@@ -0,0 +1,829 @@
+### Please use this command to compile this file into the parent `js` directory:
+    coffee --no-header -w -o ../ -c bika.lims.site.coffee
+###
+
+
+class window.SiteView
+
+  load: =>
+    console.debug "SiteView::load"
+
+    # load translations
+    jarn.i18n.loadCatalog 'bika'
+    @_ = window.jarn.i18n.MessageFactory('bika')
+
+    # initialze the loading spinner
+    @init_spinner()
+
+    # initialze the client add overlay
+    @init_client_add_overlay()
+
+    # initialze the client combogrid
+    @init_client_combogrid()
+
+    # initialze datepickers
+    @init_datepickers()
+
+    # initialze reference definition selection
+    @init_referencedefinition()
+
+    # initialze department filtering
+    @init_department_filtering()
+
+    # bind the event handler to the elements
+    @bind_eventhandler()
+
+    # allowed keys for numeric fields
+    @allowed_keys = [
+      8,    # backspace
+      9,    # tab
+      13,   # enter
+      35,   # end
+      36,   # home
+      37,   # left arrow
+      39,   # right arrow
+      46,   # delete - We don't support the del key in Opera because del == . == 46.
+      44,   # ,
+      60,   # <
+      62,   # >
+      45,   # -
+      69,   # E
+      101,  # e,
+      61    # =
+    ]
+
+    # The name of the department filter cookies
+    @department_filter_cookie = "filter_by_department_info"
+    @department_filter_disabled_cookie = "dep_filter_disabled"
```

Comment:
Remind me to cleanup all these department filtering things spreaded all over the code someday. If I cannot remove that cookies dance, at least I want their names to be consistent (e.g. always use `_senaite_core_` prefix...)